smalltalksrc/VMMaker/StackInterpreter.class.st

```diff
@@ -2624,55 +2624,61 @@ StackInterpreter >> binarySmiCompInlinePrimitive: primIndex [
 StackInterpreter >> booleanCheatFalseSistaV1 [
 	"cheat the interpreter out of the pleasure of handling the next bytecode IFF it is a jump-on-boolean. Which it is, often enough when the current bytecode is something like bytecodePrimEqual"
 
-	<sharedCodeInCase: #bytecodePrimGreaterThanSistaV1>
+	"<sharedCodeInCase: #bytecodePrimGreaterThanSistaV1>"
 	self druidIgnore: [
 		| bytecode offset |
 		bytecode := self fetchByte. "assume next bytecode is jumpIfFalse (99%)"
-		self pop: 2.
 		(bytecode < 199 and: [ bytecode > 191 ]) ifTrue: [ "short jumpIfFalse"
-			^ self jump: bytecode - 191 ].
+			self jump: bytecode - 191.
+			^ nil ].
 
 		bytecode = 239 ifTrue: [ "long jumpIfFalse"
 			offset := self fetchByte.
-			^ self jump: offset ].
+			self jump: offset.
+			^ nil ].
 
 		"not followed by a jumpIfFalse; (un)do instruction fetch and push boolean result"
 		currentBytecode := bytecode ].
-	self push: objectMemory falseObject
+	^ objectMemory falseObject
 ]
 
 { #category : #utilities }
 StackInterpreter >> booleanCheatSistaV1: cond [
 	"cheat the interpreter out of the pleasure of handling the next bytecode IFF it is a jump-on-boolean. Which it is, often enough when the current bytecode is something like bytecodePrimEqual"
-	<inline: true>
 
-	cond
-		ifTrue: [self booleanCheatTrueSistaV1]
-		ifFalse: [self booleanCheatFalseSistaV1]
+	<inline: true>
+	| resultBoolean |
+	self pop: 2.
+	resultBoolean := cond
+		ifTrue: [ self booleanCheatTrueSistaV1 ]
+		ifFalse: [ self booleanCheatFalseSistaV1 ].
+	resultBoolean ifNotNil: [ self push: resultBoolean ]
 ]
 
 { #category : #utilities }
 StackInterpreter >> booleanCheatTrueSistaV1 [
 	"cheat the interpreter out of the pleasure of handling the next bytecode IFF it is a jump-on-boolean. Which it is, often enough when the current bytecode is something like bytecodePrimEqual"
 
-	<sharedCodeInCase: #bytecodePrimLessThanSistaV1>
-	self pop: 2.
+	"<sharedCodeInCase: #bytecodePrimLessThanSistaV1>"
```

Comment:
This is not necessary? Or did you comment the line for a test?

Reply:
The pragma is necessary for slang translation.
But with my changes such a translation is not valid anymore.
We should probably discuss this better.